examples/llm/src/models/layers/attention.py

```diff
@@ -12,10 +12,21 @@
 from composer.algorithms.low_precision_layernorm.low_precision_layernorm import \
     LPLayerNorm
 from einops import rearrange
-from omegaconf import DictConfig
 from torch import nn
 
 
+def _reset_is_causal(num_query_tokens: int, num_key_tokens: int,
+                     original_is_causal: bool):
+    if num_query_tokens != num_key_tokens:
```

Comment:
```suggestion
    if original_is_causal and num_query_tokens != num_key_tokens:
```

defaults to false if original_is_causal is false
avoids `raise NotImplementedError` if we don't need to check it.